secretsclient/lazy_ssh_agent.go

```diff
@@ -0,0 +1,107 @@
+package secretsclient
+
+import (
+	"net"
+
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/crypto/ssh/agent"
+)
+
+type lazySSHAgent struct {
+	sockPath string
+	sshAgent agent.ExtendedAgent
+}
```

Comment:
Wonder if we can leverage https://golang.org/pkg/sync/#Once for this. I don't know if it simplifies this lazy proxy code though.

Reply:
I don't think we would be able to handle errors using this.